source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,42 @@
 class UrlsController < ApplicationController
+
+  def index
+    @urls = Url.all
+  end
+
+  def new
+  end
+
+  def create
+    @url = Url.new(url_params)
+
+    if @url.save
+      redirect_to '/'
+    else
+      @urls = Url.all
+      render 'index'
+    end
+  end
+
+  def destroy
+    @url = Url.find(params[:id])
+    @url.destroy
+
+    redirect_to '/'
+  end
+
+  def click_link
+    @url = Url.find(params[:format])
+
+    new_count = @url.click_count + 1
+    @url.update_attribute(:click_count, new_count)
```

Comment:
There actually is a `increment_counter` class method that updates the database for you as well:
```ruby
Url.increment_counter :click_count, @url.id
```
It's unfortunate it doesn't appear to be available as an instance method but I moved it to a `click!` instance method on `Url`.